addons/lunch/models/lunch_supplier.py

```diff
@@ -0,0 +1,172 @@
+        return time.max
+    fractional, integral = math.modf(hours)
+    res = time(int(integral), int(float_round(60 * fractional, precision_digits=0)), 0)
+    if tz:
+        res.replace(tzinfo=pytz.timezone(tz))
+    return res
+
+def time_to_float(t):
+    return float_round(t.hour + t.minute/60 + t.second/3600, precision_digits=2)
+
+class LunchSupplier(models.Model):
+    _name = 'lunch.supplier'
+    _description = 'Lunch Supplier'
+
+    partner_id = fields.Many2one('res.partner', string='Linked partner', required=True)
+
+    name = fields.Char('Name', related='partner_id.name')
+    description = fields.Text('Description')
+
+    email = fields.Char(related='partner_id.email')
+    email_formatted = fields.Char(related='partner_id.email_formatted', readonly=True)
+    phone = fields.Char(related='partner_id.phone')
+    street = fields.Char(related='partner_id.street')
+    street2 = fields.Char(related='partner_id.street2')
+    zip = fields.Char(related='partner_id.zip')
+    city = fields.Char(related='partner_id.city')
+    state_id = fields.Many2one("res.country.state", related='partner_id.state_id')
+    country_id = fields.Many2one('res.country', related='partner_id.country_id')
+
+    vat = fields.Char(related='partner_id.vat')
+
+    image = fields.Binary(related='partner_id.image', readonly=False)
+    image_medium = fields.Binary(related='partner_id.image_medium', readonly=False)
+    image_small = fields.Binary(related='partner_id.image_small', readonly=False)
+
+    send_by = fields.Selection([
+        ('phone', 'Phone'),
+        ('mail', 'Email'),
+    ], 'Send Order By', default='phone')
+    automatic_email_send = fields.Boolean('Automatic Email Sending')
+    automatic_email_time = fields.Float('Hour')
+
+    recurrency = fields.Selection([('once', 'Specific Day'), ('reccurent', 'Reccurent')], 'Recurrency', default='once')
+    recurrency_from = fields.Float('From')
+    recurrency_to = fields.Float('To')
+    recurrency_date = fields.Date('Day', default=fields.Date.today())
+    recurrency_date_from = fields.Datetime('from', compute='_compute_recurrency_date_from', store=True)
+    recurrency_date_to = fields.Datetime('to', compute='_compute_recurrency_date_to', store=True)
+    recurrency_monday = fields.Boolean('Monday')
+    recurrency_tuesday = fields.Boolean('Tuesday')
+    recurrency_wednesday = fields.Boolean('Wednesday')
+    recurrency_thursday = fields.Boolean('Thursday')
+    recurrency_friday = fields.Boolean('Friday')
+    recurrency_saturday = fields.Boolean('Saturday')
+    recurrency_sunday = fields.Boolean('Sunday')
+
+    available_today = fields.Boolean('', compute='_compute_available_today', search='_search_available_today')
+
+    tz = fields.Selection(_tz_get, string='Timezone', required=True, default='UTC')
+
+    _sql_constraints = [
+        ('automatic time', 'CHECK(automatic_email_time >= 0 AND automatic_email_time <= 24)', 'Automatic Email Sending Time should be between 0 and 24'),
```

Comment:
use underscored notation for the technical name, and be more precise. Like "automatic_email_time_range"